src/Domain.LinnApps/ImportBooks/ImportBookService.cs

```diff
@@ -0,0 +1,123 @@
+﻿namespace Linn.Stores.Domain.LinnApps.ImportBooks
+{
+    using System.Collections.Generic;
+    using Linn.Common.Persistence;
+    using Linn.Stores.Domain.LinnApps.Parts;
+
+    public class ImportBookService : IImportBookService
+    {
+        private readonly IRepository<ImportBookInvoiceDetail, ImportBookInvoiceDetailKey> InvoiceDetailRepository;
+        private readonly IRepository<ImportBookOrderDetail, ImportBookOrderDetailKey> OrderDetailRepository;
+        private readonly IRepository<ImportBookPostEntry, ImportBookPostEntryKey> PostEntryRepository;
+
+
+        public ImportBookService(
+            IRepository<ImportBookInvoiceDetail, ImportBookInvoiceDetailKey> invoiceDetailRepository,
+            IRepository<ImportBookOrderDetail, ImportBookOrderDetailKey> orderDetailRepository,
+            IRepository<ImportBookPostEntry, ImportBookPostEntryKey> postEntryRepository)
+        {
+            this.InvoiceDetailRepository = invoiceDetailRepository;
+            this.OrderDetailRepository = orderDetailRepository;
+            this.PostEntryRepository = postEntryRepository;
+        }
+
+        public void Update(
+            ImportBook from,
+            IEnumerable<ImportBookInvoiceDetail> invoiceDetails,
+            IEnumerable<ImportBookOrderDetail> orderDetails,
+            IEnumerable<ImportBookPostEntry> postEntries)
```

Comment:
I'd be tempted to just chuck all these child lists in with a whole ImportBook to parameter...
Not sure why though. Maybe just a little cleaner

Reply:
I read a bit further down and realised your updating all the entities simple files in the face and then doing the child list updates in here, so I guess it makes sense to pass the entity in and then the new child lists in after all